### PR TITLE
cni-plugins: add compat for aws-k8s-cni

### PR DESCRIPTION
**Pull request by sanghanan**
- Added missing [`tap`](https://www.cni.dev/plugins/current/main/tap/) and [`vrf`](https://www.cni.dev/plugins/current/meta/vrf/) to the plugin groups.
- Added a compat package for `aws-k8s-cni-init`. All the plugins need to be copied into the `/init` directory. The entrypoint script then copies them to the host, so we can't use `ln`. See: https://github.com/chainguard-dev/image-requests/issues/4138
- Added simple `--version` test
- Refactored the plugins list into `vars`

### CODE REVIEW

File: cni-plugins.yaml [hunk 1]
```diff
@@ -96,6 +100,10 @@ subpackages:
       - runs: |
           mkdir -p "${{targets.subpkgdir}}"/usr/bin
           cp bin/${{range.key}} "${{targets.subpkgdir}}"/usr/bin
+    test:
+      pipeline:
+        - runs: |
+            ${{range.key}} --version
   - range: plugins
```
Comment (EyeCantCU) [inline cni-plugins.yaml:107]: nit: let's add spaces between these subpackages

File: cni-plugins.yaml [hunk 2]
```diff
@@ -105,6 +113,31 @@ subpackages:
       - runs: |
           mkdir -p "${{targets.subpkgdir}}"/opt/cni/bin
           ln -s /usr/bin/${{range.key}} "${{targets.subpkgdir}}"/opt/cni/bin/${{range.key}}
+  - name: "cni-plugins-aws-k8s-compat"
+    description: "Compatibility package for aws-k8s-cni"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/init
+          for f in bandwidth \
```
Comment (EyeCantCU) [inline cni-plugins.yaml:121]: We should place all of these in a var above and loop over that

Comment (EyeCantCU) [reply cni-plugins.yaml:121]: Example with PostgreSQL: https://github.com/wolfi-dev/os/blob/2f02bc0659d420eab283f8553856a450f1ab9b4d/postgresql-16.yaml#L88

Comment (sanghanan) [reply cni-plugins.yaml:121]: that's neat, will change it!

File: cni-plugins.yaml (hunk 1 above)
Comment (sanghanan) [reply cni-plugins.yaml:102]: `yam` removes the space between `subpackages`, so CI fails 😞 
adding the space would definitely help readability though

File: cni-plugins.yaml (hunk 2 above)
Comment (sanghanan) [reply cni-plugins.yaml]: ended up doing this for the other subpackages too, definitely makes it easier to maintain

File: cni-plugins.yaml (hunk 1 above)
Comment (sanghanan) [reply cni-plugins.yaml:98]: The `.yam.yaml` accounts for a gap between the subpackages: https://github.com/wolfi-dev/os/blob/main/.yam.yaml#L3, which makes this confusing

Comment (EyeCantCU) [reply cni-plugins.yaml:98]: That is incredibly strange, but given everything else is in order, let's get this through

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**